web/source/kmwnative.ts

```diff
@@ -26,11 +26,13 @@ if(!window['keyman']['initialized']) {
 
     /**
      * Set default device options
-     * @param {Object}  opt device options object
+     * @param {OptionType}  opt device options object
      */
-    keymanweb.setDefaultDeviceOptions=function(opt) {
+    keymanweb.setDefaultDeviceOptions = function(opt : com.keyman.OptionType) {
       // Element attachment type
-      if(opt['attachType'] == '') opt['attachType'] = (device.touchable ? 'manual' : 'auto');
+      if (opt['attachType'] == '' || opt['attachType'] === undefined) {
```

Comment:
```suggestion
      if(!opt['attachType']) {
```

Covers both cases, the missing `null` case, _and_ is more compact.